### PR TITLE
SIP2-153 : Update to Java 17 edge-sip2

### PR DESCRIPTION
**Pull request by nirmalr-shah**
[SIP2-153](https://issues.folio.org/browse/SIP2-153) : _Update to Java 17 edge-sip2_

## Purpose
 Update the module to Java 17 and identify issues.
 https://wiki.folio.org/display/TC/JDK+17+and+Java+17

## Approach
 Changed the java sdk version to 17 wherever applicable

### CODE REVIEW

File: Dockerfile
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply Dockerfile:1]: done

File: pom.xml
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply pom.xml:1]: fixed

[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply pom.xml:1]: added

[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply pom.xml:1]: upgraded!!

File: src/main/java/org/folio/edge/sip2/repositories/FeeFinesRepository.java [hunk 1]
```diff
@@ -355,16 +354,16 @@ public Future<FeePaidResponse> performFeePaidCommand(FeePaid feePaid, SessionDat
               JsonObject accts = resource.getResource();
               final JsonArray acctList = accts.getJsonArray("accounts");
               Float acctTotal = totalAmount(acctList);
-              BigDecimal bdAmountPaid = new BigDecimal(moneyFormatter.format(amountPaid));
-              BigDecimal bdAmountTotal = new BigDecimal(moneyFormatter.format(acctTotal));
+              BigDecimal bdAmountPaid = new BigDecimal(Float.toString(amountPaid), moneyFormat);
+              BigDecimal bdAmountTotal = new BigDecimal(Float.toString(acctTotal), moneyFormat);
```
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply src/main/java/org/folio/edge/sip2/repositories/FeeFinesRepository.java]: removed

File: pom.xml [hunk 2]
```diff
@@ -80,9 +82,9 @@
       <dependency>
         <groupId>org.mockito</groupId>
         <artifactId>mockito-bom</artifactId>
-        <version>4.3.1</version>
+        <version>4.6.1</version>
         <type>pom</type>
-        <scope>import</scope>
+        <scope>test</scope>
```
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply pom.xml]: reverted as earlier

File: pom.xml [hunk 3]
```diff
@@ -176,11 +178,19 @@
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
+      <version>4.10.0</version>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-junit-jupiter</artifactId>
+      <version>4.3.1</version>
```
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply pom.xml]: reverted as earlier

File: pom.xml [hunk 4]
```diff
@@ -176,11 +178,19 @@
     <dependency>
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
+      <version>4.10.0</version>
```
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply pom.xml]: reverted as earlier

File: pom.xml [hunk 5]
```diff
@@ -80,9 +82,9 @@
       <dependency>
         <groupId>org.mockito</groupId>
         <artifactId>mockito-bom</artifactId>
-        <version>4.3.1</version>
+        <version>4.6.1</version>
```
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply pom.xml]: reverted as earlier

File: src/main/java/org/folio/edge/sip2/repositories/FeeFinesRepository.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nirmalr-shah) [reply src/main/java/org/folio/edge/sip2/repositories/FeeFinesRepository.java]: Done. please re-review

**Review by Vignesh-kalyanasundaram — APPROVED** [review]

**Pull request merged**